compiler/vmgen.nim

```diff
@@ -2074,6 +2081,12 @@ proc gen(c: PCtx; n: PNode; dest: var TDest; flags: TGenFlags = {}) =
     checkCanEval(c, n)
     case s.kind
     of skVar, skForVar, skTemp, skLet, skParam, skResult:
+      if sfCompileTime in s.flags and c.mode == emRepl and
+          gfCompileTimeInit notin flags:
+        genGlobalVarInit(c, s, newTreeI(nkIdentDefs, s.info,
```

Comment:
This makes it as broken as JS backend. I can also make runtime access to compileTime variables work in the nimscript (since it shares the same VM) at the cost of breaking compileTime at the forloop.